src/server/game/Movement/MotionMaster.cpp

```diff
@@ -468,6 +470,29 @@ void MotionMaster::MoveSmoothPath(uint32 pointId, Movement::PointsArray const& p
     //MovePoint(EVENT_CHARGE_PREPATH, pos, false);
 }
 
+void MotionMaster::MoveAlongSplineChain(uint32 pointId, uint32 dbChainId, bool walk)
+{
+    Creature* owner = _owner->ToCreature();
+    ASSERT(owner && "MoveAlongSplineChain (w/ chain from DB) may only be called on creatures!");
+    SplineChain const& chain = sScriptSystemMgr->GetSplineChain(owner, dbChainId);
+    if (chain.empty())
+    {
+        TC_LOG_ERROR("misc", "MotionMaster::MoveAlongSplineChain: creature with entry %u tried to walk along non-existing spline chain with DB id %u.", owner->GetEntry(), dbChainId);
+        return;
+    }
+    MoveAlongSplineChain(pointId, chain, walk);
+}
+
+void MotionMaster::MoveAlongSplineChain(uint32 pointId, SplineChain const& chain, bool walk)
+{
+    Mutate(new SplineChainMovementGenerator(pointId, chain, walk), MOTION_SLOT_CONTROLLED);
+}
+
+void MotionMaster::ResumeSplineChain(SplineChainResumeInfo const& info)
+{
+    Mutate(new SplineChainMovementGenerator(info), MOTION_SLOT_CONTROLLED);
```

Comment:
intended use is "enter combat, pause splinechain, store info, on exit combat, leash then resume once home reached".